src/iccpd/src/iccp_cmd_show.c

```diff
@@ -449,6 +489,116 @@ int iccp_local_if_dump(char * *buf,  int *num, int mclag_id)
                     return EXEC_TYPE_FAILED;
             }
         }
+
+        if (csm->peer_link_if) {
+
+            lif_peer = csm->peer_link_if;
+
+            memset(&mclagd_lif, 0, sizeof(struct mclagd_local_if));
+
+            mclagd_lif.ifindex = lif_peer->ifindex;
+
+            if (lif_peer->type == IF_T_UNKNOW)
+                memcpy(mclagd_lif.type, "Unknown", 6);
+            else if (lif_peer->type == IF_T_PORT)
+                memcpy(mclagd_lif.type, "Ethernet", 8);
+            else if (lif_peer->type == IF_T_PORT_CHANNEL)
+                memcpy(mclagd_lif.type, "PortChannel", 11);
+
+            memcpy(mclagd_lif.name, lif_peer->name, MAX_L_PORT_NAME);
+            memcpy(mclagd_lif.mac_addr, lif_peer->mac_addr, ETHER_ADDR_LEN);
+
+            if (lif_peer->state == PORT_STATE_UP)
+                memcpy(mclagd_lif.state, "Up", 2);
```

Comment:
What is the allocated space for "state"?  Really should not manually count the string length...

Reply:
@gechiang : Thanks , Corrected.!